appmap/_implementation/event.py

```diff
@@ -249,32 +250,39 @@ def __init__(self, message_parameters):
             self.message.append(message_object)
 
 
+def filter_http_headers(headers):
+    """Filter out HTTP headers which shouldn't go into appmap events.
+    Return None if none left."""
+    headers = {
+        k: v for k, v in headers.items()
+        if k not in ['Content-Type', 'Authorization', 'Host', 'User-Agent']
+    }
+    return headers if len(headers) > 0 else None
+
+
 class HttpRequestEvent(MessageEvent):
```

Comment:
By the way, we don't have real message parameters for Flask - https://github.com/applandinc/appmap-python/blob/897fd4ed609c23ee3f0c76c23747dc1db1a0729b/appmap/flask.py#L70

Although we do for Django - https://github.com/applandinc/appmap-python/blob/be049598c95d7536685201dc09f3be79ae50d12a/appmap/django.py#L74

Without message parameters, Swagger gen won't work too well.